File: src/main/java/tyrael/excel/CsvWriter.java
package tyrael.excel;

import java.util.List;

import tyrael.file.FileUtil;

public class CsvWriter {
    private final String file;

    public CsvWriter(String file) {
        this.file = file;
    }

    public void write(List<String> list){
        FileUtil.write(file, list);
    }
}

File: src/main/java/aaron/excel/CSVWriter.kt
package aaron.excel

import java.io.File
import java.io.PrintStream

class CSVWriter(private val filePath: String) {
    fun write(list: List<String?>) {
        val file = File(filePath)
        file.parentFile.mkdirs()

        // if file does not exist, then create it
        if (!file.exists()) {
            file.createNewFile()
        }

        val ps = PrintStream(file)
        val uft8bom = byteArrayOf(0xef.toByte(), 0xbb.toByte(), 0xbf.toByte())
        ps.write(uft8bom)
        for (s in list) {
            ps.println(s)
        }
        ps.close()
    }
}